src/table_builder.cc

```diff
@@ -60,6 +60,37 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
       AppendInternalKey(&index_key, ikey);
       base_builder_->Add(index_key, index_value);
     }
+  } else if (ikey.type == kTypeBlobIndex && cf_options_.level_merge &&
+             target_level_ >= merge_level_) {
+    // we merge value to new blob file
+    BlobIndex index;
+    Slice copy = value;
+    status_ = index.DecodeFrom(&copy);
+    if (!ok()) {
+      return;
+    }
+    auto storage = blob_storage_.lock();
+    assert(storage != nullptr);
+    auto blob_file = storage->FindFile(index.file_number).lock();
+    if (ShouldMerge(blob_file)) {
+      BlobRecord record;
+      PinnableSlice buffer;
+      Status s = storage->Get(ReadOptions(), index, &record, &buffer);
+      if (s.ok()) {
+        std::string index_value;
+        AddBlob(ikey.user_key, record.value, &index_value);
+        if (ok()) {
+          std::string index_key;
+          ikey.type = kTypeBlobIndex;
+          AppendInternalKey(&index_key, ikey);
+          base_builder_->Add(index_key, index_value);
+        }
+      } else {
```

Comment:
verbose `base_builder_->Add(key, value)`

Reply:
agree. let's refactor the logic here a bit to avoid multiple `base_builder_->Add(key, value)` calls.